packages/flutter/lib/src/material/tabs.dart

```diff
@@ -2019,6 +2019,12 @@ class _TabBarViewState extends State<TabBarView> {
         viewportFraction: widget.viewportFraction,
       );
     } else {
+      // Sometimes, TabBarView gets initialized before rendering.
```

Comment:
How would that happen though?

Reply:
For example, I have a page A with `tabView` and page B with a button to control it. If put both of them in memory, the router will only render page B, but page maybe put into memory also. In that case, if you click the button, the tab view on the page A won't be able to get `viewportDimension ` as it was not rendered. For that reason I believe adding guard on `jumpToPage` or `animateToPage` will be the solution. Because the controller's index will be changed after clicking the button, the controller can reflect the UI when the page A is ready to load.